### ISSUE TITLE
Ruby process hangs at 100% CPU when including `rmagick` gem on macOS 12.3.

### ISSUE THREAD

**Issue by joshuapinter** [rmagick/rmagick#1320]
### Description
I (accidentally) updated to the latest macOS 12.3 beta and now when I start my Rails app or even run a `rake -T` to list rake tasks, it hangs. Looking at the processes, it shows `ruby` running at 100%. It stays like this indefinitely until I use a `kill -9` to kill the process.

The issue is temporarily "resolved" by commenting out `gem "rmagick"` from my `Gemfile`.

I tried re-installing `rmagick`, `imagemagick` and even `ruby` but they all exhibit the same behaviour.

Not sure how to diagnose this further or if it is related to the latest macOS version or not.

### Steps to Reproduce
Include `gem "rmagick"` in `Gemfile` and try and run `rake -T` or `rails c` or `rails s`.

### System Configuration
- ImageMagick version: 7.1.0-20
- RMagick version: 4.2.4
- Ruby version: 2.7.5 (also tested with 2.7.4 and 3.1.0 with the same issue)
- Environment (Operating system, version and so on): macOS 12.3 Beta (was working fine previously on macOS 12.2)
- Additional information: I'm honestly a little lucky I found the connection to `rmagick` at all. I'm not sure how else to diagnose and proceed from here. Happy to help diagnose this further to help nail this down. 

Thanks!

**Comment by joshuapinter** [comment]
👀 Let me know if there's any further diagnostics I can do to help narrow this down. Thanks!

**Comment by Watson1978** [comment]
I'll give it a try at the weekend.
At least, could you try to update ImageMagick 7.1.0-20 to the latest version? because it might be broken (ref. https://github.com/rmagick/rmagick/pull/1318) .

**Comment by joshuapinter** [comment]
I upgraded to `imagemagick` 7.1.0-22, removed and re-installed `rmagick`, and still getting the same issue of the command (`bundle exec rake -T`) hanging with the `ruby` process pinning the CPU core. Once I comment out `gem "rmagick"`, the command runs perfectly. Strange.

Let me know if you can reproduce this on your end or if there is something else I can try on my end to help nail it down. Thanks again.

**Comment by Watson1978** [comment]
I can reproduce this problem with ImageMagick 7.1.0-22 on my Apple M1 Max with macOS 12.3 Beta.
Looks for me that it load the library continuously forever.

However, this problem doesn't seem to happen with ImageMagick 6.

```
$ system_profiler SPSoftwareDataType
Software:

    System Software Overview:

      System Version: macOS 12.3 (21E5196i)
      Kernel Version: Darwin 21.4.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: M1Max
      User Name: Watson (watson)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 7 days 20:13

$ brew list | grep imagemagick
imagemagick@6

$(brew --prefix imagemagick@6)/bin/Magick-config --version
6.9.12-37 Q16

```

I guess we may not be able to handle this as RMagck problem.
Can you report the issue to ImageMagick team too?

**Comment by dlemstra** [comment]
This doesnt sound like an ImageMagick issue when the process hangs when the rmagick library is loaded. Sounds more like a ruby or apple tooling issue. Don't think we can help with this.

**Comment by joshuapinter** [comment]
@Watson1978 Thank you so much for going through the process upgrading to the beta version of macOS on a M1 system and testing this out. The fact that you were able to reproduce this eases my mind, regardless if it's something we can resolve with `rmagick`. Much appreciated.

I imagine/hope that when 12.3 comes out officially this will either get resolved by itself or should get some more attention from either `imagemagick` or the Apple dev tools communities.  

Do you want me to create the same Issue with `imagemagick`? Or will it carry more weight coming from one of your?

Let me know and thanks again.

**Comment by dlemstra** [comment]
I am one of the maintainers of ImageMagick @joshuapinter and this really does not sound like an ImageMagick issue. If ImageMagick works on your machine then something happens when it gets used by RMagick and we have no idea what could cause this.

**Comment by joshuapinter** [comment]
I didn't know that @dlemstra, thanks for the update! Okay, well, maybe I'll re-open this for the time being and wait to see what the root of the cause is. Thanks again and let me know if there anything further I can do to help diagnose and scope this issue down.

**Comment by webhoernchen** [comment]
I I have a similar problem.
However, in my project, when Rmagick is loaded, the shell commands no longer work.
The command hangs forever.

Example: `echo 'test'`

My system:
MacOS 12.3 beta 3 Apple M1
ruby: 2.6.9 | 3.0.2
rmagick 2.4.4

Rmagick itself works

**Comment by Watson1978** [comment]
Seem this issue should be solved with macOS 12.3 beta 4.

@joshuapinter Can you confirm it?

**Comment by joshuapinter** [comment]
@Watson1978 Yup, confirmed! Fixed in macOS 12.3 Beta 4 (Build 21E5222a). 

They also fixed Reminders synchronizing, which broke in the previous build, so that's great. 

I won't be going back to beta releases of macOS anytime soon. Thanks for all the help with this, at least knowing I wasn't alone here.
